pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -43,100 +43,96 @@ def _convert_to_su2(U, return_global_phase=False):
-            [ 0.53396245-0.10177564j,  0.76279558-0.35024096j]
-        ])
-
-    For PennyLane devices that cannot natively implement ``QubitUnitary``, we
-    can instead recover a ``Rot`` gate that implements the same operation, up
-    to a global phase:
-
-    >>> decomp = zyz_decomposition(U, 0)
+    >>> U = np.array([[-0.28829348-0.78829734j,  0.30364367+0.45085995j],
+    ...               [ 0.53396245-0.10177564j,  0.76279558-0.35024096j]])
+    >>> decomp = zyz_decomposition(U, 0, return_global_phase=True)
     >>> decomp
-    [Rot(-0.24209529417800013, 1.14938178234275, 1.7330581433950871, wires=[0])]
+    [RZ(tensor(-0.2420953, requires_grad=True), wires=[0]),
+    RY(tensor(1.14938178, requires_grad=True), wires=[0]),
+    RZ(tensor(1.73305815, requires_grad=True), wires=[0]),
+    (0.38469215914523336-0.9230449299422961j)*(Identity(wires=[0]))]
     """
 
     # Cast to batched format for more consistent code
     U = math.expand_dims(U, axis=0) if len(U.shape) == 2 else U
 
-    U = _convert_to_su2(U)
-
-    # If U is only one unitary and its value is not abstract, we can include a conditional
-    # statement that will check if the off-diagonal elements are 0; if so, just use one RZ
-    if len(U) == 1 and not math.is_abstract(U[0]):
-        if math.allclose(U[0, 0, 1], 0.0):
-            return [qml.RZ(2 * math.angle(U[0, 1, 1]), wires=wire)]
+    U_det1, alphas = _convert_to_su2(U, return_global_phase=True)
 
     # For batched U or single U with non-zero off-diagonal, compute the
-    # Rot operator decomposition instead
-    off_diagonal_elements = math.clip(math.abs(U[:, 0, 1]), 0, 1)
+    # normal decomposition instead
+    off_diagonal_elements = math.clip(math.abs(U_det1[:, 0, 1]), 0, 1)
     thetas = 2 * math.arcsin(off_diagonal_elements)
 
     # Compute phi and omega from the angles of the top row; use atan2 to keep
     # the angle within -np.pi and np.pi, and add very small value to the real
     # part to avoid division by zero.
     epsilon = 1e-64
     angles_U00 = math.arctan2(
-        math.imag(U[:, 0, 0]),
-        math.real(U[:, 0, 0]) + epsilon,
+        math.imag(U_det1[:, 0, 0]),
+        math.real(U_det1[:, 0, 0]) + epsilon,
     )
     angles_U10 = math.arctan2(
-        math.imag(U[:, 1, 0]),
-        math.real(U[:, 1, 0]) + epsilon,
+        math.imag(U_det1[:, 1, 0]),
+        math.real(U_det1[:, 1, 0]) + epsilon,
     )
 
     phis = -angles_U10 - angles_U00
     omegas = angles_U10 - angles_U00
 
-    phis, thetas, omegas = map(math.squeeze, [phis, thetas, omegas])
+    # Wrap angles to range [-\pi, \pi]
+    phis = (phis + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
+    omegas = (omegas + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
 
-    return [qml.Rot(phis, thetas, omegas, wires=wire)]
+    phis, thetas, omegas, alphas = map(math.squeeze, [phis, thetas, omegas, alphas])
+
+    Operations = [qml.RZ(phis, wire), qml.RY(thetas, wire), qml.RZ(omegas, wire)]
```

Comment:
```suggestion
    operations = [qml.RZ(phis, wire), qml.RY(thetas, wire), qml.RZ(omegas, wire)]
```
We do not user upper case at the start of a variable

Reply:
Fixed in next commit.